### PR TITLE
Add definition for bidirectional text. Refine First-strong def. Use g…

### PR DESCRIPTION
**Pull request by r12a**
…roup in config.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/string-meta/pull/52.html" title="Last updated on Sep 4, 2020, 2:50 PM UTC (8c2239a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/string-meta/52/19ee17d...8c2239a.html" title="Last updated on Sep 4, 2020, 2:50 PM UTC (8c2239a)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -114,8 +104,9 @@ <h3>Terminology</h3>
...
   <p><dfn>LTR</dfn> stands for "left-to-right" and refers to the inline base direction of left-to-right [[UAX9]]. This is the base text direction used by languages whose starting character progression begins on the left side of the page in horizontal text. It's used for scripts such as Latin, Cyrillic, Devanagari, and many others.</p>
   <p><dfn>RTL</dfn> stands for "right-to-left" and refers to the inline base direction of right-to-left [[UAX9]]. This is the base text direction used by languages whose starting character progression begins on the right side of the page in horizontal text. It's used for a variety of scripts which include Arabic, Hebrew, N'Ko, Adlam, Thaana, and Syriac among others.</p>
+<p><dfn>Bidirectional text</dfn> is a term used to refer to text that mixes runs of both LTR and RTL text inline. It is common for text written in Arabic, Hebrew, etc. to contain short runs of Latin text, and several of the scripts that are written predominantly right-to-left write their numbers from left-to-right. Bidirectional text of this kind is the source of many of the difficulties when dealing with RTL scripts.</p>
```
Comment (aphillips) [inline index.html]: This needs more work, I think.

* I would introduce "bidi" has a shorthand for this term. Or do you mean to use this term distinctly?
* _is a term_ is redundant in a definition
* the examples of Arabic/Hebrew, while amended with "etc", may be too specifically focused on those scripts. Also it's possible that they'll be read as meaning the Arabic/Hebrew languages
* _short runs of Latin text_ is overly specific. It's by far the most common case, of course. But we should allow for other inclusions.
* _bidirectional text of this kind_ reuses the term in a way that implies there are other kinds of bidirectional text.

Perhaps:

**Bidirectional text** (often referred to as "_bidi_" for short) refers to text that includes runs of both LTR and RTL text inline. It is common for right-to-left scripts, such as Arabic and Hebrew, to contain short runs of left-to-right text (most commonly in the Latin script), and several of the scripts that are predominantly right-to-left write their numbers from left-to-right. Bidirectional text is the source of many of the difficulties when dealing with RTL scripts.

File: index.html [hunk 2]
```diff
@@ -114,8 +104,9 @@ <h3>Terminology</h3>
...
+<p> <dfn>First-strong detection</dfn> is an algorithm that looks for the first strongly-directional character in a string that is not inside a nested isolate, and then uses that to guess at the appropriate base direction for the string as a whole. Unicode code points are associated with properties relating to text direction: generally, letters in right-to-left scripts such as Arabic and Hebrew have a strong RTL direction, whereas Latin and Han characters have a strong LTR direction. Other characters, such as punctuation, only have a weak intrinsic directionality, and the actual directionality is determined according to the context in which they are found.</p>
```
Comment (aphillips) [inline index.html]: _that is not inside a nested isolate_ is kind of mysterious here. I know what you mean, but it took me a bit to grok it. Perhaps:

**First-strong detection** is an algorithm that looks for the first strongly-directional character in a string (while ignoring isolates), and then uses that...

**Comment by r12a** [comment]
Good suggestions. Reworked the text.

File: index.html [hunk 3]
```diff
@@ -104,9 +104,9 @@ <h3>Terminology</h3>
...
-<p><dfn>Bidirectional text</dfn> is a term used to refer to text that mixes runs of both LTR and RTL text inline. It is common for text written in Arabic, Hebrew, etc. to contain short runs of Latin text, and several of the scripts that are written predominantly right-to-left write their numbers from left-to-right. Bidirectional text of this kind is the source of many of the difficulties when dealing with RTL scripts.</p>
+<p><dfn>Bidirectional text</dfn> (often referred to as &quot;bidi&quot; for short) refers to text that mixes runs of both LTR and RTL text inline. It is common for right-to-left scripts, such as Arabic and Hebrew, to contain short runs of left-to-right text (most commonly in the Latin script), and several of the scripts that are predominantly right-to-left display  numbers from left-to-right. Bidirectional text is the source of many of the difficulties when dealing with RTL scripts.</p>
```
Comment (aphillips) [inline index.html:107]: Add `data-lt`, i.e. `<dfn data-lt="bidirectional text|bidi">` ?

**Review by aphillips — COMMENTED** [review]
Looks good to me, pace the one small comment.

**Comment by r12a** [comment]
I think that would then need to read 'bidi text', no?  I demurred on that previously, but for the definition we'd need to get it straight.

**Comment by aphillips** [comment]
So you're trying to distinguish "bidirectional text" from the concept of "bidi" here? I kind of got that from the definition--that you didn't mean "bidi" in the generic sense but more in the sense of "mixed direction text", but won't it be confusing? 

On some level, all text is "potentially bidirectional". And fully RTL text is technically "unidirectional". But generically at least we tend to talk about RTL text as "bidi" because LTR tends to be the default presentation. If you want to have a term for text that happens to actually *be* bidirectional, perhaps use the term "mixed direction text" instead to keep from being confusing? Or do you **also** intend to make this term usable outside of string-meta?

**Comment by r12a** [comment]
Hmmmmm. I'm not trying to distinguish 'bidirectional text' from 'bidi' at all.  I've never considered that 'bidi' or 'bidi text' is a synonym for 'rtl-text'.  That certainly seems to me far too sloppy a definition to define in string-meta (or elsewhere).  When i refer to 'bidi text', concept or instance, i mean text that runs, or has the potential to run, in more than one inline direction.  If i refer to 'bidi issues', i mean those issues that are produced or could be produced by mixing text directions, whether in  a RTL or a LTR script.  I think that using 'bidi text' when you meant 'RTL text' is certainly likely to lead to confusion.

**Comment by r12a** [comment]
Agreed to close because definitions belong in the Glossary.

**Pull request closed without merging**